subworkflows/local/utils_nfcore_proteinfamilies_pipeline/main.nf

```diff
@@ -224,3 +206,29 @@ def methodsDescriptionText(mqc_methods_yaml) {
     return description_html.toString()
 }
 
+//
+// Validate that two folders (HMMs and MSAs for update) contain the same number of files and matching base filenames
+//
+def validateMatchingFolders(channel1, channel2) {
+    // Fetch the contents of the channels
+    channel1
+        .join(channel2)
+        .map { meta, folder1, folder2 ->
+            def files1 = folder1.listFiles()
+            def files2 = folder2.listFiles()
+
+            // Check if the number of files matches
+            if (files1.size() != files2.size()) {
+                error("Folder mismatch: ${folder1} has ${files1.size()} files, but ${folder2} has ${files2.size()} files.")
```

Comment:
```suggestion
                error("[nf-core/proteinfamilies] ERROR: Folder mismatch: ${folder1} has ${files1.size()} files, but ${folder2} has ${files2.size()} files.")
```

I always like to add a prefix like this as it helps tell users is it a pipeline error, vs a tool error, vs a proper pipeline error. I would add something like this to all `error()` calls you've made.